File: SiHyeon/week2/다트 게임.cpp
#include <string>
#include <cmath>
#include <vector>

using namespace std;

int solution(string dartResult) {
    int answer = 0;
    int index = 0;
    vector<int> score;
    while (index < dartResult.size() - 1)
    {
        int value = dartResult[index++] - '0';
        if (dartResult[index] == '0') //다음에 0이 오면 10 // 맘에 안드는 방법이지만..
        {
            value = 10;
            index++;
        }
        char bonus = dartResult[index++];
        int point = 0;
        //보너스 계산
        if (bonus == 'S')
            point = value;
        else if (bonus == 'D')
            point = pow(value, 2);
        else
            point = pow(value, 3);
        //옵션계산
        if (dartResult[index] == '*')
        {
            point *= 2;
            if (!score.empty()) //벡터가 있으면 이전 점수에도 옵션을 반영
                score[score.size() - 1] *= 2;
            index++;
        }
        else if (dartResult[index] == '#')
        {
            point *= -1;
            index++;
        }
        score.push_back(point);
    }
    for (auto v : score)
        answer += v;
    return answer;
}

File: hyeonwook/week2/다트게임.java
package prgmrs;

import java.io.IOException;

public class 다트게임 {
    public static int solution(String dartResult) {
        int answer = 0;
        int [] score = new int[3];
        int index=0;
        String str = "";
        char [] dart = new char[dartResult.length()];
        for (int i = 0; i < dartResult.length(); i++) {
            dart[i] = dartResult.charAt(i);
        }
        for (int i = 0; i < dart.length; i++) {

            if(dart[i]>='0' && dart[i]<='9'){
                str += String.valueOf(dartResult.charAt(i));

            }
            else if(dart[i]=='S' ||dart[i]=='D'||dart[i]=='T' ){
                int num = Integer.parseInt(str);
                if(dart[i]=='S'){
                    score[index++]+= (int)Math.pow(num,1) ;
                }
                else if(dart[i]=='D'){
                    score[index++]+= (int)Math.pow(num,2);
                }
                else if(dart[i]=='T'){
                    score[index++]+= (int)Math.pow(num,3);
                }
                str ="";
            } else {
                if(dart[i]=='*'){
                    score[index-1]*=2;
                    if(index-2>=0) score[index-2]*=2;
                }
                else {
                    score[index-1]*=(-1);
                }
            }
        }
            answer= score[0]+score[1]+score[2];

        return answer;
    }
    public static void main(String[] args) throws IOException {

        System.out.println(solution("1S*2T*3S"));


    }
}
